src/api_v2/models/user.rs

```diff
@@ -0,0 +1,73 @@
+use std::time::{SystemTime, UNIX_EPOCH};
+use crate::api_v2::Error;
+
+/// ```
+/// if User::uid == 0{
+///     .. //bot user
+/// } else {
+///     .. //user
+/// };
+/// ```
+///
+pub struct User {
+    pub uid: i32,
+    pub time: u64,
+    pub access_token: String,
+    pub refresh_token: String,
+}
+
+pub trait UserSave {
+    fn save(&self, user: &User);
+}
+
+impl User {
+    pub fn next_time(&mut self, add_time: u64) {
+        let c;
+        if let Ok(duration) = SystemTime::now().duration_since(UNIX_EPOCH) {
+            c = duration.as_millis();
+        } else {
+            c = 0;
+        }
+        self.time = c as u64 + add_time;
+    }
+    
+    fn new() -> Self{
```

Comment:
You can implement default trait for this, instead of a `new` and a `new_bot`. https://doc.rust-lang.org/std/default/trait.Default.html